src/Events/CouponRedeemed.php

```diff
@@ -11,7 +12,14 @@ class CouponRedeemed
     use Dispatchable;
     use InteractsWithSockets;
 
-    public function __construct(public Coupon $coupon)
+    public function __construct(
+        public Coupon $coupon,
+        /**
+         * The order on which the coupon was redeemed. Should be made mandatory
+         * in v8.
+         */ 
+        public ?Order $order = null
```

Comment:
This event will only be dispatched by Simple Commerce (people's apps shouldn't be dispatching SC events). We're probably fine for this to be mandatory.

Reply:
Updated the code as requested and edited then re-ran the tests :)